### PR TITLE
Add comprehensive test suite, fix minor issues, rename function and file names to match approved verbs

### PR DESCRIPTION
**Pull request by Raphire**
# Summary by CodeRabbit

- Added a comprehensive Pester 5 test suite (28 new `Tests/*.Tests.ps1` files) covering registry backup/restore (snapshot conversion/validation), `.reg` parsing, app-removal flows, settings/apps FileIO, GUI/CLI orchestration, user/profile resolution, feature invoke/undo paths, start-menu replacement/restore, restore-backup dialog helpers, and telemetry scheduled-task cataloging.
- Introduced a local test runner (`Scripts/Run-Tests.ps1`) with an optional `-Bootstrap` flag to install Pester 5.9.0 (current user) when missing, then run the suite and fail the run when tests don’t pass, have failures, or report zero tests.
- Added GitHub Actions CI (`.github/workflows/tests.yml`) to run the same test command (`./Scripts/Run-Tests.ps1 -Bootstrap`) on pull requests, pushes to `master`, and manual dispatch using Windows PowerShell 5.1 with a 15-minute timeout (read-only contents permissions, no credential persistence).
- Standardized PowerShell verb-noun/kebab-case naming across the codebase, renaming many exported functions and updating call sites (widespread `Get-*`/`Set-*`/`Invoke-*`/`Test-*`, `Add-Parameter`, `Import-*`, `Save-*`, `Wait-ForKeyPress`, `Write-CliHeader`, `Invoke-DoEvents`), including wiring updates in `Win11Debloat.ps1`.
- Improved core robustness in registry and parsing logic to satisfy the new test expectations (stricter registry value kind/data validation, safer hex/binary conversions, clearer restore/backup failure behavior, and normalization fixes).
- Added/extended supporting helpers to support the test suite and standardized flows (telemetry scheduled-task helpers, registry-path deletion helper, restore dialog feature-list helpers, store search suggestion `Set-*` APIs, plus additional verb-noun wrapper/rename updates for apply/undo orchestration).

### CODE REVIEW

File: Tests/Invoke-Changes.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,351 @@
+BeforeAll {
+    function Import-RegistryFile { param($Message, $RegistryFile) }
```
[the remark this answers is not in the document]
Comment (Raphire) [reply Tests/Invoke-Changes.Tests.ps1]: This should be fixed now, please confirm

File: Tests/Registry-BackupCreation.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+Describe 'Get-RegistryBackupPayload' {
+    BeforeEach {
+        Mock Get-RegistryBackedFeatures { param($Features) @($Features | Where-Object RegistryKey) }
+        Mock Get-RegistryBackupCapturePlans { @([PSCustomObject]@{ Path = 'HKEY_CURRENT_USER\Software\Example' }) }
+        Mock Get-RegistrySnapshotsForBackup { @([PSCustomObject]@{ Path = 'HKEY_CURRENT_USER\Software\Example'; Exists = $true }) }
+        Mock Get-RegistryBackupTargetDescription { 'CurrentUser:Tester' }
+        $env:COMPUTERNAME = 'TestComputer'
+    }
```
[the remark this answers is not in the document]
Comment (Raphire) [reply Tests/Registry-BackupCreation.Tests.ps1:40]: This should be fixed now, please confirm

File: Tests/Settings-And-Apps-FileIO.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        $apps[0].AppId | Should -Be 'Two.App'
+    }
+
+    It 'loads presets and preserves their ID arrays' {
+        $presets = @(Import-AppPresetsFromJson)
+
+        $presets | Should -HaveCount 1
+        $presets[0].Name | Should -Be 'Minimal'
+        $presets[0].AppIds | Should -Be 'One.App'
+    }
```
[the remark this answers is not in the document]
Comment (Raphire) [reply Tests/Settings-And-Apps-FileIO.Tests.ps1:177]: This should be fixed now, please confirm

File: Scripts/Features/Registry-BackupValidation.ps1 [hunk 4]
```diff
@@ -314,6 +315,65 @@ function Test-RegistrySnapshotAgainstAllowList {
...
+            return $true
+        }
+        ([Microsoft.Win32.RegistryValueKind]::MultiString) {
+            foreach ($item in @($Data)) {
+                if ($item -isnot [string]) { return $false }
+            }
+            return $true
+        }
+        ([Microsoft.Win32.RegistryValueKind]::None) { return ($null -eq $Data) }
+        default { return ($null -eq $Data -or $Data -is [string]) }
```
[the remark this answers is not in the document]
Comment (Raphire) [reply Scripts/Features/Registry-BackupValidation.ps1:388]: This should be fixed now, please confirm

File: Scripts/Helpers/Get-FriendlyRegistryBackupTarget.ps1 [hunk 5]
```diff
@@ -1,4 +1,4 @@
-function GetFriendlyRegistryBackupTarget {
+function Get-FriendlyRegistryBackupTarget {
```
[the remark this answers is not in the document]
Comment (Raphire) [reply Scripts/Helpers/Get-FriendlyRegistryBackupTarget.ps1:5]: This should be fixed now, please confirm
